test/tests/interfacekernels/1d_interface/ADMatreaction_1D_steady.i

```diff
@@ -0,0 +1,172 @@
+# Steady-state test for the ADMatInterfaceReaction kernel.
+#
+# Specie M transport from domain 1 (0<=x<=1) to domain 2 (1<x<=2),
+# u and v are concentrations in domain 1 and domain 2.
+#
+# Diffusion in both domains can be described by Ficks law and diffusion
+# kernel is applied.
+#
+# Specie M has different diffusity in different domains, here set as D1=4, D2=2.
+#
+# Dirichlet boundary conditions are applied, i.e., u(0)=1, v(2)=0
+#
+# At the interface consider the following
+#
+# (a) Fluxes are matched from both domains (InterfaceDiffusion kernel)
+#
+# (b) First-order reaction is R = kf*u - kb*v
```

Comment:
Are you sure the fluxes are matched? The condition you are implicitly setting at the interface for the u variable is this: $-D_u \nabla u \cdot \hat{n} = -D_v \nabla v \cdot \hat{n} + k_f u - k_b v$, so I hope that the fluxes do not match

Reply:
Can you comment on this?

Reply:
I agree with your assessment of the implicit condition at the interface. It looks like a Robin BC on both sides to me. 

The flux at both sides of the interface should be matched, with some specified offset, which may change and may also be linked to a material property. The - and + side fluxes should negate each other and the u-v value terms should also negate, though I suppose there might be other potential values of the solution if the flux/value specifications conflict with each other. 

Clearly I'm missing something in what you're trying to tell me, but I'd also point out that this is verbatim copied from the `reaction_1D_steady.i` test, so whatever the resolution is, we'll need to change it there too.

Reply:
I think we should delete these lines and replace them by the interface condition we are weakly enforcing which is $-D_u \nabla u \cdot \hat{n} = -D_v \nabla v \cdot \hat{n} + k_f u - k_b v$ Basically we'll want to match the equation we put into the documentation

Reply:
I'll write the weakly enforced boundary condition we're applying, but I really want to be explicit about exactly how we're applying that condition, so I'd recommend not deleting the explanation of the constituent parts. The class this addition is based off of was my first class I really dug into in the moose framework, and it took me way too long to understand what each piece was doing, and what I needed to put in to make sure the physics was right. Of course, the documentation was much less robust back then.

Reply:
I had to throw some `SideDiffusiveFluxIntegral` postprocessors on the non-AD test locally to test this out because I did not believe it. But indeed the diffusive fluxes are equivalent like you said! So instead of simply $-D_u \nabla u \cdot \hat{n} = -D_v \nabla v \cdot \hat{n} + k_f u - k_b v$ we have $-D_u \nabla u\cdot\hat{n} = -D_v \nabla v \cdot \hat{n}$ and $k_f u = k_b v$. It's not entirely clear in my head what is constraining the fluxes to be the same on either side of the interface but my feeling is the requirement for a constant gradient on either subdomain.

But yea, I guess we can keep this comment!

Reply:
One thing you might consider is writing: `R = kf*u - kb*v = 0`